wicker/core/column_files.py

```diff
@@ -85,22 +91,22 @@ class ColumnBytesFileWriter:
 
     def __init__(
         self,
-        storage: S3DataStorage = S3DataStorage(),
-        s3_path_factory: S3PathFactory = S3PathFactory(),
+        storage: AbstractDataStorage = S3DataStorage(),
+        path_factory: WickerPathFactory = S3PathFactory(),
         target_file_size: Optional[int] = None,
         target_file_rowgroup_size: Optional[int] = None,
         dataset_name: str = None,
     ) -> None:
         """Create a ColumnBytesFileWriter
 
         :param storage: storage client to use, defaults to S3DataStorage()
-        :param s3_path_factory: path factory to use, defaults to S3PathFactory()
+        :param path_factory: path factory to use, defaults to S3PathFactory()
         :param target_file_size: If set, open a new binary file when a column file gets larger than this many bytes.
         :param target_file_rowgroup_size: If set, open a new binary file when a column file has more rows than this
         :param dataset_name: dataset name, defaults to None
         """
         self.storage = storage
-        self.s3_path_factory = s3_path_factory
```

Comment:
I don't think this is backward compatible with predecessors.

Reply:
Do you mind @aalavian if we make this backwards-incompatible variable renaming? The `ColumnBytesFileWriter` class should really be able to write column bytes files to more places than just S3. IMHO this is an important change that is worth the renaming.

Although this code change is indeed backwards-incompatible, I made searches in all of our Woven GitHub repositories and also a public GitHub-wide search for references to the `ColumnBytesFileWriter` class to make sure this change should not really impact anyone.

Technically, clients are not supposed to be interacting directly with `ColumnBytesFileWriter` anyways as they should be using subclasses of [AbstractDataPersistor](https://github.com/woven-planet/wicker/blob/main/wicker/core/persistance.py#L21), but I understand your general backwards compatibility point.

Reply:
@aalavian now we've added a specific `self.s3_path_factory` property to make sure we have this backwards-compatibility property set just in case there are any references to the `ColumnBytesFileWriter` class across all of our codebases that I missed in my searches. In that case, the code will find this property successfully.